elastic/integrations
# [elasticsearch] update for GC log level

## What does this PR do?

Elasticsearch will add a log level to its JVM logs to allow users to help them detect errors / warnings more easily in elastic/elasticsearch#92382. With this commit we detect this new field if present and continue to recognize the prior log format which does not contain a log level.

## Checklist

- [x] I have tested my changes to prove they are working fine and are backword compatible
- [x] Added tests to validate my work

## How to this PR was tested
- Added log level to the [ingest pipeline](https://github.com/elastic/integrations/blob/main/packages/elasticsearch/data_stream/gc/elasticsearch/ingest_pipeline/default.yml#L36)
- Built an [ES docker image](https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc) to make sure it has the [latest changes](https://github.com/elastic/elasticsearch/pull/92382) since the image is not pushed yet to [docker hub](https://www.docker.elastic.co/r/elasticsearch?show_snapshots=true).
- Used the local built image in the [docker-compose file.](https://github.com/elastic/integrations/blob/main/packages/elasticsearch/_dev/deploy/docker/docker-compose.yml#L4)
- Started a complete ES stack using `elastic-package`.
- Used [elastic-package](https://github.com/elastic/elastic-package#elastic-package-service) to start up the [log generation service](https://github.com/elastic/integrations/blob/main/packages/elasticsearch/_dev/deploy/docker/docker-compose.yml#L26) 
  - From inside the [elasticsearch](https://github.com/elastic/integrations/tree/main/packages/elasticsearch) package folder run:
     - `elastic-package service up -v`
     - logs are going to be generated under `~/.elastic-package/tmp/service_logs`
  - Checked the logs to make sure the log level is written in the logs:
     -  Before => `[2023-01-10T15:03:54.104+0000][209][gc,start    ] GC(9) Pause Cleanup`
     -  After    => `[2023-01-11T12:03:54.104+0000][209][info ][gc,start    ] GC(9) Pause Cleanup`
  - From kibana installed the Elasticsearch integration and checked ingest pipeline to make sure that the new log level field is reflected in the pipeline
  
<img width="1720" alt="Screenshot 2023-01-13 at 16 40 25" src="https://user-images.githubusercontent.com/11225826/212346556-f6d05cfb-3e34-4978-82ad-1d9ff77baa0b.png">

  - From discover you can check the log.level field to make sure that it reads the new values written to the logs
<img width="1712" alt="Screenshot 2023-01-13 at 13 52 33" src="https://user-images.githubusercontent.com/11225826/212332883-ecb015ef-3a05-4fa7-8007-fb4b2215dff1.png">
<img width="1727" alt="Screenshot 2023-01-13 at 13 52 45" src="https://user-images.githubusercontent.com/11225826/212332889-57d4c17a-8209-44d0-a4da-d1d7513995ab.png">

## Related issues

- subtask of [#92382](https://github.com/elastic/elasticsearch/pull/92382)
